harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -778,26 +1168,49 @@ def _load(self, load_path: pathlib.Path) -> None:
             logging.warning("The checkpoint has no random state to restore.")
 
         callback_state = checkpoint.get("callbacks", {})
-        for name in self.callbacks:
+        for name in self._callbacks:
             if name in callback_state:
-                self.callbacks[name].load_state_dict(callback_state[name])
-            elif util.is_overridden(self.callbacks[name].load_state_dict, pytorch.PyTorchCallback):
+                self._callbacks[name].load_state_dict(callback_state[name])
+            elif util.is_overridden(self._callbacks[name].load_state_dict, pytorch.PyTorchCallback):
                 logging.warning(
                     f"Callback '{name}' implements load_state_dict(), but no callback state "
                     "was found for that name when restoring from checkpoint. This "
                     "callback will be initialized from scratch"
                 )
 
-        # Load workload sequencer state.
-        wlsq_path = load_path.joinpath("workload_sequencer.pkl")
-        if self.wlsq is not None and wlsq_path.exists():
-            with wlsq_path.open("rb") as f:
-                self.wlsq.load_state(pickle.load(f))
+        save_path = load_path.joinpath("trial_state.pkl")
+        if save_path.exists():
+            with save_path.open("rb") as f:
+                self._load_state(pickle.load(f))
+        else:
+            # Support legacy save states
```

Comment:
Question.  With regard to supporting legacy save states, why not do a conversion inside here, the legacy-case-detection block?  Why put the legacy conversion inside the common-codepath-block (`_load_state()`)?

Reply:
if you refresh, i added a separate codepath _load_wlsq_state for this.